src/v/raft/log_eviction_stm.h

```diff
@@ -56,16 +56,18 @@ class log_eviction_stm {
     }
 
 private:
-    ss::future<> handle_deletion_notification(model::offset);
     void monitor_log_eviction();
+    void process_log_eviction();
+    ss::future<> housekeeping();
+    ss::future<> do_process_log_eviction(model::offset);
 
     consensus* _raft;
     ss::logger& _logger;
     ss::lw_shared_ptr<storage::stm_manager> _stm_manager;
     ss::abort_source& _as;
     ss::gate _gate;
-    model::offset _previous_eviction_offset;
     model::offset _requested_eviction_offset;
+    std::deque<model::offset> _eviction_offsets;
```

Comment:
Yup this was definitely a place I wanted feedback in. It doesn't hurt to do this, we will still process the events in the order they would have arrived in before these changes were implemented, but now there are other truncation events via delete-records requests as well.

IMO the strategy should be truncate at the max offset from all of the events that have been obtained. The smaller offsets can be thrown away.

Reply:
Yeah, I went with this method though because there would need to be no locking, and when an offset for an event that is lower then visible_offset is observed, there will be no work to be done.